### PR TITLE
Improve Kernel documentation

### PR DESCRIPTION
**Pull request by samnang**
Hi,

I just updated and added some more examples for Kernel's methods.
Let's help to check it out.

Thanks,
Samnang

### CODE REVIEW

File: complex.c [hunk 1]
```diff
@@ -477,7 +477,11 @@ f_complex_new2(VALUE klass, VALUE x, VALUE y)
  * call-seq:
  *    Complex(x[, y])  ->  numeric
  *
- * Returns x+i*y;
+ * Returns (x+i*y) as a <code>Complex</code>; Where x is real part, 
+ * y is imaginary part and i is imaginary unit.
```
Comment (steveklabnik) [inline complex.c]: is "imaginary unit" better than 'the square root of negative one?'

Comment (samnang) [inline complex.c]: Either one is fine, but I want to keep it consistants with description in [Complex](http://www.ruby-doc.org/core-1.9.3/Complex.html).

Comment (phiggins) [inline complex.c]: I think all of 'real part', 'imaginary part' and 'imaginary unit' should have 'the' before them.

File: load.c [hunk 2]
```diff
@@ -358,14 +358,17 @@ rb_load_protect(VALUE fname, int wrap, int *state)
  *  call-seq:
  *     load(filename, wrap=false)   -> true
  *
- *  Loads and executes the Ruby
- *  program in the file _filename_. If the filename does not
+ *  Loads and executes the Ruby program in the file _filename_ 
+ *  every time it's called. If the filename does not
```
Comment (phiggins) [inline load.c]: I'd remove the contraction "it's" in favor of "it is".

File: object.c [hunk 3]
```diff
@@ -2529,7 +2529,10 @@ rb_Array(VALUE val)
  *  Returns <i>arg</i> as an <code>Array</code>. First tries to call
  *  <i>arg</i><code>.to_ary</code>, then <i>arg</i><code>.to_a</code>.
  *
- *     Array(1..5)   #=> [1, 2, 3, 4, 5]
+ *     Array(1..5)     #=> [1, 2, 3, 4, 5]
+ *     Array([1, 2])   #=> [1, 2]
+ *     Array(1)        #=> [1]
+ *     Array(nil)      #=> []
```
Comment (phiggins) [inline object.c]: You might want to add in the common gotcha of passing a hash to Array.

File: random.c [hunk 4]
```diff
@@ -765,10 +765,22 @@ random_load(VALUE obj, VALUE dump)
...
  *  seeds the generator using a combination of the time, the
  *  process id, and a sequence number. (This is also the behavior if
- *  <code>Kernel::rand</code> is called without previously calling
+ *  <code>Kernel#rand</code> is called without previously calling
  *  <code>srand</code>, but without the sequence.) By setting the seed
- *  to a known value, scripts can be made deterministic during testing.
- *  The previous seed value is returned. Also see <code>Kernel::rand</code>.
+ *  to a known value, scripts can be made deterministic during testing, and
+ *  even in different applications sequence of generated randoms numbers 
+ *  are the same. The previous seed value is returned.
```
Comment (phiggins) [inline random.c]: I'd change this:

> and even in different applications sequence of generated randoms numbers are the same.

to:

> and even different applications or systems will generate the same sequence of random numbers.

File: rational.c [hunk 5]
```diff
@@ -548,7 +548,11 @@ f_rational_new_no_reduce2(VALUE klass, VALUE x, VALUE y)
  * call-seq:
  *    Rational(x[, y])  ->  numeric
  *
- * Returns x/y;
+ * Returns (x/y) as a <code>Rational</code>; Where x is numerator and
+ * y is denominator.
```
Comment (phiggins) [inline rational.c]: I'd change this to something like:

> Returns (x/y) as a <code>Rational</code> with the numerator x and the denominator y.

Do x and y need to be marked up somehow, possibly surrounded by "+" symbols?

File: rational.c [hunk 6]
```diff
@@ -548,7 +548,11 @@ f_rational_new_no_reduce2(VALUE klass, VALUE x, VALUE y)
  * call-seq:
  *    Rational(x[, y])  ->  numeric
```
Comment (phiggins) [inline rational.c]: This should probably be changed from numeric to Rational.

**Comment by phiggins** [comment]
Great work @samnang!

**Comment by samnang** [comment]
Thank @phiggins, overall comments look good. I have pushed the updates.

File: complex.c [hunk 7]
```diff
@@ -475,9 +475,14 @@ f_complex_new2(VALUE klass, VALUE x, VALUE y)
 
 /*
  * call-seq:
- *    Complex(x[, y])  ->  numeric
+ *    Complex(x[, y])  ->  complex
  *
- * Returns x+i*y;
+ * Returns <code>(x+i*y)</code> as a Complex;
+ * Where x is the real part, y is the imaginary part and
```
Comment (phiggins) [inline complex.c]: Sorry I didn't notice before, but this reads a little awkwardly. I'd suggest removing the semicolon sort of like: "as a Complex instance where x is the real part ...".

**Comment by phiggins** [comment]
Other than the last note I made, it looks great! Let's get one more person to read over this. :thumbsup:

File: complex.c (hunk 7 above)
Comment (samnang) [inline complex.c]: Yes, I agree.

**Pull request merged**